### PR TITLE
fix: upgrade slip10 to fix esm

### PR DESCRIPTION
**Pull request by mvayngrib**
## Test Plan

thanks @chalker for the issue and test plan 

```
import { fromMasterSeed as bip32FromMasterSeed } from '@exodus/bip32'
import SLIP10 from '@exodus/slip10'

const MAP_KDF = Object.freeze({
  BIP32: bip32FromMasterSeed,
  SLIP10: SLIP10.fromSeed,
})

console.log(Object.entries(MAP_KDF))
```

before: `[ [ 'BIP32', [Function: fromMasterSeed] ], [ 'SLIP10', undefined ] ]`
after: `[ [ 'BIP32', [Function: fromMasterSeed] ], [ 'SLIP10', [Function: fromSeed] ] ]`

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -32,6 +32,7 @@ const untranspiledModulePatterns = [
   '@exodus/dependency-preprocessors',
   '@exodus/osmosis-*',
   '@exodus/key-utils',
+  '@exodus/slip10',
```
Comment (mvayngrib) [inline jest.config.js:35]: jest yells at me otherwise about not allowed to use import outside of a module

**Comment by ChALkeR** [comment]
Will re-test

**Review by ChALkeR — APPROVED** [review]
let's land this

File: yarn.lock [hunk 2]
```diff
@@ -2148,6 +2148,17 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@exodus/slip10@npm:^2.0.0":
```
Comment (fboucquez) [inline yarn.lock:2151]: can we dedup?

Comment (mvayngrib) [reply yarn.lock:2151]: no, we need to upgrade key-ids -> keychain first

**Review by diegomura — APPROVED** [review]
utACK

**Review by fboucquez — APPROVED** [review]
utACK

**Pull request merged**